backends/arm/test/passes/test_tag_io_quant_pass.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+import unittest
+
+import torch
+from executorch.backends.arm.arm_backend import generate_ethosu_compile_spec
+from executorch.backends.arm.test import common
+
+from executorch.backends.arm.test.tester.arm_tester import ArmTester
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class Linear(torch.nn.Module):
+    def __init__(
+        self,
+        in_features: int,
+        out_features: int = 3,
+        bias: bool = True,
+    ):
+        super().__init__()
+        self.inputs = (torch.ones(1, 1, 1, in_features),)
+        self.fc = torch.nn.Linear(
+            in_features=in_features,
+            out_features=out_features,
+            bias=bias,
+        )
+
+    def get_inputs(self):
+        return self.inputs
+
+    def forward(self, x):
+        return self.fc(x)
+
+
+class TestTagIOQuantPass(unittest.TestCase):
+
+    def _tosa_BI_u55_pipeline(self, module: torch.nn.Module):
+        (
+            ArmTester(
+                module,
+                inputs=module.get_inputs(),
+                compile_spec=generate_ethosu_compile_spec(
+                    "ethos-u55-128",
+                    permute_memory_to_nhwc=True,
+                    quantize_io=True,
+                ),
+            )
+            .quantize()
+            .export()
+            .to_edge()
+            .partition()
+            .check_count(
```

Comment:
These should be added even without the pass right? Through the `quantize()` assuming static quant is the default? Trying to understand how are we testing the pass.

Reply:
Yes, but they are by default consumed by our delegates, meaning that without the pass there are no quant/dequant outside our delegates.

We would like to test the pass (and future) passes in a more isolated way, preferably by calling the pass directly from the tester. Currently we are testing the whole pipeline and are making sure that we get one q and dq outside of the delegate.